src/Controller/DashboardController.php

```diff
@@ -95,6 +95,32 @@ public function deleteAction(EntityManagerInterface $em, Action $action)
         return $this->redirectToRoute('dashboard_home');
     }
 
+    /**
+     * @Route("/copy/action/{action}", name="copy_action")
+     *
+     * @param EntityManagerInterface $em
+     * @param Action $action
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function copyAction(EntityManagerInterface $em, Action $action)
+    {
+        if ($this->getUser()->getId() !== $action->getUser()->getId()) {
```

Comment:
Il faudrait plutot vérifier que la personne ne surveille pas déjà ce magasin pas seulement que c'est le sien

$action = $em->getRepository('App:Action')->findOneBy(['store' => $action->getStore(), 'storeId' => $action->getStoreId()]);

Reply:
et user => $this->getUser()